### PR TITLE
[C-1563] Fetch and download all paginated track favorites

### PR DESCRIPTION
**Pull request by amendelsohn**
### Description

Fetch the small version of all the user's favorites so we don't have to paginate the lineup to know what to download.

Todo:
- [ ] Sort the API results by date favorited so the on-screen tracks go first (could hack it by using the saves lineup tracks)

### Dragons

Performance with large numbers of tracks

### How Has This Been Tested?

Simulator

Todo:
- [ ] Test on device

### Feature Flags ###

`OFFLINE_MODE_ENABLED`

### CODE REVIEW

File: packages/mobile/src/hooks/useFetchAllFavoritedTrackIds.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+export const useFetchAllFavoritedTrackIds = () => {
+  const currentUserId = useSelector(getUserId)
+  const isOfflineModeEnabled = useIsOfflineModeEnabled()
+  return useAsync(async () => {
+    if (!isOfflineModeEnabled) return []
+    let trackIds: number[] = []
+    let loadMore = true
+    let offset = 0
+    while (loadMore) {
+      const url = apiClient.makeUrl(
```
Comment (amendelsohn) [inline packages/mobile/src/hooks/useFetchAllFavoritedTrackIds.ts:21]: I tried doing this using the common `AudiusAPIClient` but was having trouble. It always used the `/full/` endpoint even when I passed in the `PathType.VersionPath`. Probably missed something or built it wrong; not sure.

Comment (sliptype) [reply packages/mobile/src/hooks/useFetchAllFavoritedTrackIds.ts:21]: Strange, I feel like it should work. Did you look at one of the other non-full endpoints as a reference? https://github.com/AudiusProject/audius-client/blob/main/packages/common/src/services/audius-api-client/AudiusAPIClient.ts#L1277

File: packages/mobile/src/services/offline-downloader/offline-downloader.ts [hunk 2]
```diff
@@ -47,7 +47,7 @@ export const downloadCollection = async (
 ) => {
   store.dispatch(addCollection(collection))
   persistCollectionDownloadStatus(collection, true)
-  store.dispatch(batchStartDownload(trackIds.map(toString)))
+  store.dispatch(batchStartDownload(trackIds.map((id) => id.toString())))
```
Comment (sliptype) [inline packages/mobile/src/services/offline-downloader/offline-downloader.ts:50]: This is a fun one

File: packages/mobile/src/services/offline-downloader/offline-storage.ts [hunk 3]
```diff
@@ -67,12 +67,14 @@ export const listTracks = async (): Promise<string[]> => {
...
   trackId: string
 ): Promise<UserTrackMetadata> => {
-  const trackJson = await readFile(getLocalTrackJsonPath(trackId))
   try {
+    const trackJson = await readFile(getLocalTrackJsonPath(trackId))
     return JSON.parse(trackJson)
   } catch (e) {
-    console.error(e)
-    return e
+    if (e instanceof SyntaxError) {
```
Comment (sliptype) [inline packages/mobile/src/services/offline-downloader/offline-storage.ts:74]: Is this when the file doesn't exist?

**Review by sliptype — APPROVED** [review]

File: packages/mobile/src/hooks/useFetchAllFavoritedTrackIds.ts (hunk 1 above)
Comment (amendelsohn) [reply packages/mobile/src/hooks/useFetchAllFavoritedTrackIds.ts:21]: Yeah pretty sure I had it just like that. I'm gonna merge this and circle back to it. This works for now and there's some more critical feature work still left on my plate.

**Pull request merged**